content/docs/thinking-in-react.md

```diff
@@ -31,118 +31,119 @@ Our JSON API returns some data that looks like this:
+## Քայլ 5. ավելացնել տվյալների հակառակ ուղղություն {#step-5-add-inverse-data-flow}
 
-## Step 5: Add Inverse Data Flow {#step-5-add-inverse-data-flow}
+<p data-height="600" data-theme-id="0" data-slug-hash="LzWZvb" data-default-tab="js,result" data-user="rohan10" data-embed-version="2" data-pen-title="Thinking In React: Step 5" class="codepen">Նայիր Pen-ը <a href="https://codepen.io/gaearon/pen/LzWZvb">React մտածելակերպ. քայլ 5</a> <a href="https://codepen.io">CodePen-ում</a>։</p>
 
-<p data-height="600" data-theme-id="0" data-slug-hash="LzWZvb" data-default-tab="js,result" data-user="rohan10" data-embed-version="2" data-pen-title="Thinking In React: Step 5" class="codepen">See the Pen <a href="https://codepen.io/gaearon/pen/LzWZvb">Thinking In React: Step 5</a> on <a href="https://codepen.io">CodePen</a>.</p>
+Մինչ այս, մենք կառուցել ենք հավելված, որն արտապատկերում է կախված հիերարխիայով դեպի ներքև հոսող props-ից ու state-ից։ Հիմա ժամանակն է ապահովել տվյալների հոսք մյուս ուղղությունով. հիերարխիայի խորքում գտնվող `SearchBar`-ի կոմպոնենտները կարիք ունեն `FilterableProductTable`-ի state-ը թարմացնելու։
 
-So far, we've built an app that renders correctly as a function of props and state flowing down the hierarchy. Now it's time to support data flowing the other way: the form components deep in the hierarchy need to update the state in `FilterableProductTable`.
+React-ը դարձնում է այս տվյալների հոսքը բացահայտ, որպեսզի հեշտ լինի հասկանալ, թե ինչպես է աշխատում ձեր հավելվածը։ Բայց մեզ պետք կգա մի փոքր ավել կոդ գրել ավանդական երկկողմանի դվյալների կապվածությունը ստանալու համար։
 
-React makes this data flow explicit to make it easy to understand how your program works, but it does require a little more typing than traditional two-way data binding.
+Եթե դուք փորձեք գրել կամ սեղմել checkbox-ը օրինակի ընթացիկ տարբերակում, ապա կտեսնեք, որ React-ը անտեսում է ձեր մուտքագրումները։ Սա կանխամտածված է, քանի որ մենք `input`-ում `value` prop-ին տվել էինք `FilterableProductTable`-ից փոխանցված `state`-ի արժեքը։
 
-If you try to type or check the box in the current version of the example, you'll see that React ignores your input. This is intentional, as we've set the `value` prop of the `input` to always be equal to the `state` passed in from `FilterableProductTable`.
+Եկեք մտածենք, թե ինչ ենք ուզում տեղի ունենա։ Մենք ցանկանում ենք լինել վստահ, որ ցանկացած պահի, երբ օգտագործողը փոփոխի form-ը, մենք կթարմացնենք state-ը\` օգտագործողի մուտքագրումն արտացոլելու համար։ Քանի, որ կոմպոնենտները պետք է միայն իրենց state-ները թարմացնեն, `FilterableProductTable`-ը կփոխանցի հետկանչի ֆունկցիա `SearchBar`-ին, որը կկանչի, երբ state-ը պետք լինի թարմացնել։ Մենք կարող ենք օգտագործել `onChange` իրադարձությունը դրա մասին տեղեկանալու համար։ `FilterableProductTable`-ի կողմից փոխանցված հետկանչի ֆունկցիաները կկանչեն `setState()` և հավելվածը կլինի թարմացված։
```

Comment:
"հավելվածը կլինի թարմացված" ֊> "հավելվածը կթարմանա", էդ ուղղակի անգլում էդ ժամանակի կազմությունն ա be-ով, մեզ պետք չի